pkg/apis/core/validation/validation.go

```diff
@@ -3024,6 +3025,47 @@ func ValidatePod(pod *core.Pod) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(specPath, hugePageResources, "must use a single hugepage size in a pod spec"))
 	}
 
+	podIPsField := field.NewPath("status", "podIPs")
+
+	// all PodIPs must be valid IPs
+	for i, podIP := range pod.Status.PodIPs {
+		for _, msg := range validation.IsValidIP(podIP.IP) {
+			allErrs = append(allErrs, field.Invalid(podIPsField.Index(i), podIP.IP, msg))
+		}
+	}
+
+	// if we have more than one Pod.PodIP then
+	// - validate for dual stack
+	// - validate for duplication
+	if len(pod.Status.PodIPs) > 1 {
+		// dualstack feature must be enabled if we have more than one PodIP
+		if !utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) {
+			allErrs = append(allErrs, field.Invalid(podIPsField, pod.Status.PodIPs, "multiple PodIPs were found and IPv6DualStack feature is not enabled"))
+		}
+
+		podIPs := make([]string, 0, len(pod.Status.PodIPs))
+		for _, podIP := range pod.Status.PodIPs {
+			podIPs = append(podIPs, podIP.IP)
+		}
+
+		dualStack, err := netutils.IsDualStackIPStrings(podIPs)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(podIPsField, pod.Status.PodIPs, fmt.Sprintf("invalid PodIPs. failed to check for dual stack with error:%v", err)))
```

Comment:
This should be InternalError.  You already validated that all of the IPs are valid, so if this actually returns an error it's something we don't know about.  You could ignore error here, on that assumption, but if IsDualStackIPStrings ever changes, you would not catch it.  So catch error, throw it as an internal, sleep peacefully knowing that it won't trigger, but if it does, at least we'll have a clue.